buildpipeline/perf-pipeline.groovy

```diff
@@ -0,0 +1,390 @@
+        bat "move Microsoft.BenchView.ThroughputBenchmarks.x64.Windows_NT x64ThroughputBenchmarks"
+        bat "move Microsoft.BenchView.ThroughputBenchmarks.x86.Windows_NT x86ThroughputBenchmarks"
+
+        stash name: 'benchview-artifacts', includes: 'Microsoft.BenchView.JSONFormat/**/*'
+        stash name: 'metadata', includes: '*.json'
+        stash name: 'throughput-benchmarks-x64', includes: 'x64ThroughputBenchmarks/**/*'
+        stash name: 'throughput-benchmarks-x86', includes: 'x86ThroughputBenchmarks/**/*'
+    }
+}
+
+// TODO: use non-pgo builds for throughput?
+def innerLoopBuilds = [
+    "windows x64 pgo build": {
+        simpleNode('Windows_NT','latest') {
+            windowsBuild('x64', config, '', false)
+        }
+     },
+    "windows x86 pgo build": {
+        simpleNode('Windows_NT','latest') {
+            windowsBuild('x86', config, '', false)
+        }
+    },
+    "linux x64 pgo build": {
+        simpleNode('Ubuntu16.04','latest') {
+            linuxBuild('x64', config, '', false)
+        }
+    }
+]
+
+// Only run non-pgo builds on offical builds
+def outerLoopBuilds = [:]
+
+if (!isPR()) {
+    outerLoopBuilds = [
+        "windows x64 nopgo build": {
+            simpleNode('Windows_NT','latest') {
+                windowsBuild('x64', config, 'skiprestoreoptdata', false)
+            }
+        },
+        "windows x86 nopgo build": {
+           simpleNode('Windows_NT','latest') {
+               windowsBuild('x86', config, 'skiprestoreoptdata', false)
+           }
+        },
+        "linux x64 nopgo build": {
+           simpleNode('Ubuntu16.04','latest') {
+               linuxBuild('x64', config, 'skiprestoreoptdata', false)
+           }
+        }
+    ]
+}
+
+def baselineBuilds = [:]
+
+if (isPR()) {
+   baselineBuilds = [
+       "windows x64 pgo baseline build": {
+           simpleNode('Windows_NT','latest') {
+               windowsBuild('x64', config, '', true)
+           }
+       },
+       "windows x86 pgo baseline build": {
+           simpleNode('Windows_NT','latest') {
+               windowsBuild('x86', config, '', true)
+           }
+       },
+       "linux x64 pgo baseline build": {
+           simpleNode('Ubuntu16.04','latest') {
+               linuxBuild('x64', config, '', true)
+           }
+       }
+   ]
+}
+
+stage ('Build Product') {
+    parallel innerLoopBuilds + outerLoopBuilds + baselineBuilds
+}
+
+// Pipeline builds don't allow outside scripts (ie ArrayList.Add) if running from a script from SCM, so manually list these for now.
+// Run the main test mix on all runs (PR + official)
+
+def innerLoopTests = [:]
+
+['x64', 'x86'].each { arch ->
+    [true,false].each { isBaseline ->
+        String baseline = ""
+        if (isBaseline) {
+            baseline = " baseline"
+        }
+        if (isPR() || !isBaseline) {
+            innerLoopTests["windows ${arch} ryujit full_opt pgo${baseline} perf"] = {
+               simpleNode('Windows_NT','20170427-elevated') {
```

Comment:
Just to check: will these line up with server 2016 on ci2?

Reply:
Yes, I'm in the process of updating them